### ISSUE TITLE
make inheritance applicability explicit for `scan.json`

### ISSUE THREAD

**Issue by yarikoptic** [bids-standard/bids-specification#789]
@robertoostenveld in https://github.com/bids-standard/bids-specification/issues/538#issuecomment-669749360 mentions that "using the inheritance principle there can be a scans.json at the top level". Current inheritance principle formulation seems to suggest that too, but documentation nowhere mentions `scans.json` although it could be quite a common use case.

ATM I see only a single dataset with a (broken) scans.json 
```shell
(git)smaug:/mnt/datasets/datalad/crawl/openneuro[master]git
$> ls -ls ds00*/*scans.json
4 -rw------- 1 yoh datalad 656 Oct 23  2020 ds003242/scans.json
```
and which is 
```shell
$> cat ds003242/.bidsignore 
scans.json
```
possibly likely because prior versions of bids-validator choked on it.

So I wonder if we should add `scans.json` 
- as an explicit example to "Inheritance principle" section
- to https://github.com/bids-standard/bids-specification/blob/master/src/schema/top_level_files.yaml (attn @bids-standard/raw-electrophys-ieeg 
?

**Comment by sappelhoff** [comment]
> So I wonder if we should add scans.json to https://github.com/bids-standard/bids-specification/blob/master/src/schema/top_level_files.yaml (attn raw-electrophys-ieeg)

The [top level files](https://github.com/bids-standard/bids-specification/blob/master/src/schema/objects/top_level_files.yaml) schema file says:

> This file describes files which may appear at the top level of a dataset.
 This does not include information about whether these files are required or optional.
 For that information, see `rules/top_level_files.yaml`.

It contains files like:

- README
- LICENSE
- CHANGES
- dataset_description

however, due to inheritance many other files could be placed there ... for example a `task-rest_eeg.json` file. This is currently not part of the schema in this way. Not sure if that's a shortcoming or a good thing ... but if we were to add `scans.json` there, we'd surely have to add all those other files as well for consistency - WDYT?

> as an explicit example to "Inheritance principle" section

I feel like with the recent clarifications in https://github.com/bids-standard/bids-specification/pull/946 this may not be so crucial anymore. But if you have a good idea for an example, why not add it :+1:

**Comment by robertoostenveld** [comment]
I think the same applies to `sessions.json`.

**Comment by VisLab** [comment]
`events.json too....

**Comment by sappelhoff** [comment]
I agree with both of you Robert and Kay - but what do you think: Should we add those to the `top_level_files` schema ... or rather not, because the only time they may ever be placed there is when inheritance applies.

I can see both solutions. What's your opinion @tsalo?

**Comment by VisLab** [comment]
I am not sure what putting it in the schema would look like or what the trade-offs are.

For example, the filenames for these files can't have a `sub` entity or `ses` entity as part of its name, right?

**Comment by tsalo** [comment]
I am reticent to directly encode the inheritance principle into the schema. At the moment, `.json` is treated as an extension like any other, rather than as a special sidecar file. If we want to encode the inheritance principle, we basically need to have a separate entry for each suffix group just for `.json`s, and then we need to make just about every required entity in those entries optional.

>For example, a filename for one of these files can't have a `sub` entity or `ses` entity as part of its name, right?

It could have `sub` and `ses`, as long as it appears in `sub-X/ses-Y/sub-X_ses-Y_scans.<tsv|json>`. I suppose it could have `ses`, but not `sub`, too.

**Comment by monique2208** [comment]
I just came across this issue as we ran into the error with the legacy validator, which does not accept a `sessions.json` file at the top level. The deno validator has no problem with it. Does that mean this issue is resolved?

**Comment by yarikoptic** [comment]
<details>
<summary>FWIW since original filing the number of scans.json files among openneuro datasets grew up</summary> 

```shell
$> ls -ls ds00*/*scans.json | nl
     1	4 -rw-r----- 1 yoh datalad 656 Oct 23  2020 ds003242/scans.json
     2	4 -rw-r----- 1 yoh datalad 411 Jan 18  2022 ds003812/scans.json
     3	4 -rw------- 1 yoh datalad 339 Aug 22  2022 ds004141/scans.json
     4	4 -rw------- 1 yoh datalad 339 Aug 22  2022 ds004142/scans.json
     5	4 -rw------- 1 yoh datalad 339 Dec 19  2022 ds004331/scans.json
     6	4 -rw------- 1 yoh datalad 339 May 25  2023 ds004466/scans.json
     7	4 -rw------- 1 yoh datalad 186 Jun  5 11:49 ds004636/scans.json
     8	4 -rw------- 1 yoh datalad 339 Sep 23  2023 ds004693/scans.json
     9	4 -rw------- 1 yoh datalad 339 Apr 29 10:51 ds004920/scans.json
    10	4 -rw------- 1 yoh datalad 339 Apr 29 10:54 ds004943/scans.json
    11	4 -rw------- 1 yoh datalad 339 Apr 29 10:56 ds004956/scans.json
    12	4 -rw------- 1 yoh datalad 339 Jun  5 11:27 ds005085/scans.json
    13	4 -rw------- 1 yoh datalad 339 Jun  5 11:33 ds005123/scans.json
    14	4 -rw------- 1 yoh datalad 339 Sep 23 16:09 ds005134/scans.json
    15	4 -rw------- 1 yoh datalad 339 Sep 23 16:14 ds005238/scans.json
    16	4 -rw------- 1 yoh datalad 645 Sep 23 16:07 ds005250/scans.json
    17	4 -rw------- 1 yoh datalad 339 Sep 23 16:09 ds005365/scans.json
    18	4 -rw------- 1 yoh datalad 339 Sep 23 16:10 ds005374/scans.json
    19	4 -rw------- 1 yoh datalad 339 Sep 23 16:14 ds005454/scans.json
    20	4 -rw------- 1 yoh datalad 385 Sep 23 16:14 ds005455/scans.json
```
</details>

and only the earliest dataset has (only) scans.json explicitly .bidsignored:
```shell
$> for f in ds00*/*scans.json ; do fs=$(dirname $f)/.bidsignore; grep -l scans.json $fs; done
ds003242/.bidsignore
```

BIDS validation became indeed more schema and principles driven than before. As many pointed out, I think having `scans.json` is just a particular case of inheritance principle and should not be explicitly listed in the schema anyhow since in general should be applicable to any entity-less `{suffix}.json`.   I think we should resolve this issue by an explicit example.  Proposing a PR

@monique2208 , what version of legacy bids-validator did you use?  I searched myself into

- https://github.com/bids-standard/bids-validator/issues/742

which was presumably addressed by
 
- https://github.com/bids-standard/bids-validator/pull/1200

but didn't try

3 years ago and theoretically (and based on aforementioned openneuro datasets) should not be the case any more even with legacy one

**Comment by monique2208** [comment]
We were using v1.14, and we got the following error, 

```
1: [ERR] Files with such naming scheme are not part of BIDS specification. This error is most commonly caused by typos in file names that make them not BIDS compatible. Please consult the specification and make sure your files are named correctly. If this is not a file naming issue (for example when including files not yet covered by the BIDS specification) you should include a ".bidsignore" file in your dataset (see https://github.com/bids-standard/bids-validator#bidsignore for details). Please note that derived (processed) data should be placed in /derivatives folder and source data (such as DICOMS or behavioural logs in proprietary formats) should be placed in the /sourcedata folder. (code: 1 - NOT_INCLUDED)
		./sessions.json
			Evidence: sessions.json
	Please visit https://neurostars.org/search?q=NOT_INCLUDED for existing conversations about this issue.

```

We are switching to the newer validator soon, so it should not be problem any longer
